### PR TITLE
Q030 fix

### LINKED ISSUE #1324: Edit Q030 triggering incorrectly
Beta users are reporting that Q030, which is checking that "MSA/MD, state, county, census tract should equal a valid combination and not NA" is triggering when it should not. The edit is triggering often when none of those fields are reported NA, and these combinations are passing validity edits V290, V295 and V300 (and I've confirmed that the MSA/MD, state, county and census tract combinations are valid using the FFIEC geocoder). 

I am able to replicate the error:
image

### PR DESCRIPTION
**Pull request by nickgrippin**
This fix offloads the logic of determining the CRA status of an Institution to the filer, allowing them to confirm that the information is correct or fix the LAR with a new upload.

Closes #1324

### CODE REVIEW

File: validation/src/main/scala/hmda/validation/rules/lar/quality/Q030.scala [hunk 1]
```diff
@@ -1,48 +1,20 @@
...
-              when(geo.tract not "NA") {
-                geo is validCompleteCombination
-              }
-          } and when(geo.msa is "NA") {
-            when(geo.tract not "NA") {
-              geo is validStateCountyTractCombination
-            }
-          }
-      }
+      (lar.geography.msa not "NA") and
```
Comment (schbetsy) [inline validation/src/main/scala/hmda/validation/rules/lar/quality/Q030.scala]: This only works if the user exactly follows the FIG, using `"NA"` and not `"na"` or `"not applicable"` or `""` or anything. After Juan's #1336 is in, do we need to protect against any of those options?

File: validation/src/test/scala/hmda/validation/rules/lar/quality/Q030Spec.scala [hunk 2]
```diff
@@ -1,210 +1,33 @@
...
     }
   }
 
-  implicit class LarChecker(lar: LoanApplicationRegister) {
-    def mustFail(implicit fi: Institution) = check(lar, fi) mustBe a[Failure]
-    def mustPass(implicit fi: Institution) = check(lar, fi) mustBe a[Success]
-    def check(lar: LoanApplicationRegister, fi: Institution) = Q030.inContext(ValidationContext(Some(fi), None)).apply(lar)
+  property("fails when action taken is in range and geography is NA") {
+    forAll(larGen, Gen.choose(1, 6)) { (lar, i) =>
+      val invalidGeo = Geography("NA", "NA", "NA", "NA")
```
Comment (schbetsy) [inline validation/src/test/scala/hmda/validation/rules/lar/quality/Q030Spec.scala]: We might want a spec or two that shows what happens when only one, two, or three of the Geography fields is "NA".

File: validation/src/main/scala/hmda/validation/rules/lar/quality/Q030.scala (hunk 1 above)
Comment (nickgrippin) [reply validation/src/main/scala/hmda/validation/rules/lar/quality/Q030.scala]: [This line](https://github.com/cfpb/hmda-platform/pull/1336/files#diff-d44a9bbfd5df9409b42d4f4649baa8d2R175) in his PR means that all the values will be strictly "NA".

**Comment by schbetsy** [comment]
Code here looks good to me. Have you been working with @keileyo to double-check that the business logic is exactly what she asked for?

If so, I think we're good to merge.

**Comment by keileyo** [comment]
looks good to me

**Pull request merged**